core/Network/TLS/Types.hs

```diff
@@ -19,6 +19,8 @@ module Network.TLS.Types
 import Data.ByteString (ByteString)
 import Data.Word
 
+type HostName = String
```

Comment:
Yes I've seen it too and just replicated what already exists. But I can try to do better.

IIRC it's potentially more complex that it looks because x509 packages have this too.
If a type alias is to be exported, it could be from there and not tls.